src/material.js

```diff
@@ -49,4 +49,68 @@ export class Material extends Made.Material {
         const config = material.toJSON();
         return new Material(config);
     }
+
+    // eslint-disable-next-line class-methods-use-this
+    getBasisChecks() {
+        const { basis } = this.toJSON();
+        const checks = [];
+
+        // TODO: implement testing for rules in Material class (warnings mixin?)
+        for (let i = 0; i < basis.elements.length; i++) {
```

Comment:
May be we can simplify it a bit to eliminate nested loops, something like this:
`basis.elements.map((el) => JSON.stringify(el)).forEach((e, i, a) => {
         const index = a.indexOf(e);
         if (index !== i) checks.push(...);
     });`